Tests/scripts/jira_issues.py

```diff
@@ -62,45 +62,51 @@ def jira_color_text(text: str, color: str) -> str:
     return f"{{color:{color}}}{text}{{color}}"
 
 
+def find_jira_ticket_unresolved_transition(jira_server: JIRA, jira_issue: Issue | None):
+    unresolved_transition = None
+    #  Get the available transitions for the issue
+    transitions = jira_server.transitions(jira_issue)
+    for transition in transitions:
+        if transition['name'] == JIRA_ISSUE_UNRESOLVED_TRANSITION_NAME:
+            unresolved_transition = transition
+            break
+    return unresolved_transition['id'] if unresolved_transition else None
+
+
+def transition_jira_ticket_to_unresolved(jira_server: JIRA, jira_issue: Issue | None, unresolved_transition_id: str | None):
+    if unresolved_transition_id:
+        jira_server.transition_issue(jira_issue, unresolved_transition_id)
+
+
 def find_existing_jira_ticket(jira_server: JIRA,
                               now: datetime,
                               max_days_to_reopen: int,
                               jira_issue: Issue | None,
-                              ) -> tuple[Issue | None, Issue | None, bool]:
+                              ) -> tuple[Issue | None, Issue | None, bool, str | None]:
     link_to_issue = None
     jira_issue_to_use = None
+    unresolved_transition_id = None
     if use_existing_issue := (jira_issue is not None):
         searched_issue: Issue = jira_issue
         if searched_issue.get_field("resolution"):
             resolution_date = convert_jira_time_to_datetime(searched_issue.get_field("resolutiondate"))
             if use_existing_issue := (resolution_date
                                       and (now - resolution_date)
                                       <= timedelta(days=max_days_to_reopen)):  # type: ignore[assignment]
-
-                #  Get the available transitions for the issue
-                transitions = jira_server.transitions(searched_issue)
-
-                # Find the transition with the specified ID
-                unresolved_transition = None
-                for transition in transitions:
-                    if transition['name'] == JIRA_ISSUE_UNRESOLVED_TRANSITION_NAME:
-                        unresolved_transition = transition
-                        break
-                if unresolved_transition:
-                    jira_server.transition_issue(searched_issue, unresolved_transition['id'])
+                unresolved_transition_id = find_jira_ticket_unresolved_transition(jira_server, jira_issue)
```

Comment:
why no call this immediately before calling `transition_jira_ticket_to_unresolved` 
you're just passing the return value here for no good reason.